### PR TITLE
Current Diff between amd-common and amd-hcc

### PR DESCRIPTION
**Pull request by david-salinas**
I did a merge into amd-hcc with latest amd-common into new branch updated_amd_hcc.  This is a diff between that branch and amd-common.  This will show all of the HCC customizations we left behind in amd-hcc.

### CODE REVIEW

**Comment by aaronenyeshi** [comment]
Is this PR intended to remove all these commits? I don't think we need any of the merge commits

File: lib/Target/AMDGPU/AMDGPUTargetMachine.cpp [hunk 1]
```diff
@@ -639,8 +639,10 @@ void AMDGPUPassConfig::addIRPasses() {
   addPass(createAMDGPUOpenCLEnqueuedBlockLoweringPass());
 
   if (TM.getOptLevel() > CodeGenOpt::None) {
-    addPass(createInferAddressSpacesPass());
+    // ToDo: Fix it so that it can handle addrspacecast to private
     addPass(createAMDGPUPromoteAlloca());
+    //addPass(createLowerAllocaPass());
```
Comment (aaronenyeshi) [inline lib/Target/AMDGPU/AMDGPUTargetMachine.cpp:644]: So we don't use this LowerAllocaPass? Maybe we can just remove it completely

File: include/llvm/Transforms/Utils/FunctionImportUtils.h [hunk 2]
```diff
@@ -21,6 +21,9 @@
 namespace llvm {
 class Module;
 
+// Set to true depending on option -force-import-weak
+extern bool ForceImportWeakFlag;
```
Comment (aaronenyeshi) [inline include/llvm/Transforms/Utils/FunctionImportUtils.h:25]: @scchan Do we still need this flag, or should we look for an alternative?

File: lib/Transforms/Scalar/LowerAlloca.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
+//===-- NVPTXLowerAlloca.cpp - Make alloca to use local memory =====--===//
```
Comment (aaronenyeshi) [inline lib/Transforms/Scalar/LowerAlloca.cpp:1]: Remove this if we don't need it. This was originally from NVPTX?

File: tools/llvm-split/llvm-split.cpp [hunk 4]
```diff
@@ -40,6 +41,14 @@ static cl::opt<bool>
     PreserveLocals("preserve-locals", cl::Prefix, cl::init(false),
                    cl::desc("Split without externalizing locals"));
 
+static cl::opt<bool>
+    SplitHCCKernels("split-hcc-kernels", cl::Prefix, cl::init(false),
+                   cl::desc("Split HCC Kernels and print number of files"));
+
+static cl::opt<unsigned>
+    SplitHCCThreshold("split-hcc-threshold", cl::Prefix, cl::init(100),
+                   cl::desc("Split HCC Kernels if above kernel count threshold"));
```
Comment (aaronenyeshi) [inline tools/llvm-split/llvm-split.cpp:50]: @scchan this was originally for ThinLTO and kernel splitting. Do we still need this?

**Review by aaronenyeshi — COMMENTED** [review]
Are we trying to review this commits we need in amd-common before HCC switches to use amd-common for compiler/llvm?

**Comment by david-salinas** [comment]
> Is this PR intended to remove all these commits? I don't think we need any of the merge commits

I created this PR because we wanted to have another look at what was not brought over from amd-hcc into amd-common.

**Comment by aaronenyeshi** [comment]
Great, I'm okay with removing the ThinLTO changes that we made. I guess it should eventually be upstreamed instead

**Pull request closed without merging**